### DISCUSSION TITLE
How about discarding atoms from atom family automatically?

### DISCUSSION THREAD

**Discussion by jeiea** [pmndrs/jotai#2127] [Ideas] [answered]
Though I'm not sure. I coined the following from [this code](https://github.com/pmndrs/jotai/blob/2526039ea4da082749adc8a449c33422c53d9819/src/vanilla/utils/atomFamily.ts#L45-L47).

```typescript
const newAtom = initializeAtom(param);
const originalOnMount = newAtom.onMount;
newAtom.onMount = (set) => {
  const originalOnUnmount = originalOnMount?.(set);
  atoms.set(param, [newAtom, Date.now()]);

  return () => {
    originalOnUnmount?.();
    atoms.delete(param);
  }
}
return newAtom;
```

This is very uncertain idea, but I know [riverpod](https://riverpod.dev/) has [autoDispose provider](https://docs-v2.riverpod.dev/docs/concepts/modifiers/auto_dispose) and I was very pleased with the functionality. So I suggest about this.

**Comment by dai-shi** [comment, marked as answer]
Atoms usually exist after unmount (we don't delete atom values from the store) by design.

If you want to delete it, you can do that. There might be a pitfall, but feel free to try. It should be easy to create your own version of atomFamily, based on the simplified one. https://github.com/pmndrs/jotai/discussions/1323#discussioncomment-3303035
